src/libAtomVM/nifs.c

```diff
@@ -1460,6 +1482,76 @@ static term nifs_erlang_process_info(Context *ctx, int argc, term argv[])
     return ret;
 }
 
+static int set_system_architecture(char *buf, size_t max_len)
```

Comment:
maybe snprintf would be easier to understand

Reply:
Done.